### PR TITLE
Adding YML processing

### PR DESCRIPTION
**Pull request by gvegayon**
The pull request changes how software tools are processed. Instead of reading from an existing XLS file, it now uses yaml files which were mapped from the original XLSX file. The `README.qmd` document will use the yaml files and re-generate a `software.xlsx` file under the `data`.

This pull request includes updates to the `README.md` file to correct the counts of tools and software packages by type.

Updates to tool counts:

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L8-R15): Corrected the total number of tools from 39 to 38 and the total number of completed and work-in-progress tools from 55 to 53. Updated the count for "Parameter estimation" tools from 13 to 12.

Updates to software package counts:

* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L26-R26): Corrected the count for `r` language software packages from 19 to 18.

### CODE REVIEW

File: README_files/figure-commonmark/wordcloud-1.png
Comment (apulsipher) [inline README_files/figure-commonmark/wordcloud-1.png:1]: Why do we have a word cloud for the projects? It's jumbled and hard to parse. It would be more useful if each project had a keyword associated with it and the word cloud was generated from those. I suggest either 1) adding those keywords to each or 2) cutting this from the README and creating an issue to add keywords at a later date.

File: README.qmd [hunk 1]
```diff
@@ -10,25 +10,29 @@ This repository is used for coordinating all the software development activities
...
-    Links = "Link to web page/documentation (optional)",
-    Type="Type of tool",
-    Diseases="Relevant disease(s)"
+    Tool = "tool_name",
+    Description = "brief_description",
+    Languages = "languages",
+    Repo = "github_repo_new_or_old_if_existing_one",
+    Contact = "name_of_developer_maintainer_or_key_contact",
+    Email = "email_of_developer_maintainer_or_key_contact",
+    Links = "link_to_web_page_documentation_optional",
```
Comment (apulsipher) [inline README.qmd:33]: I have mixed feelings about these column names. On one hand, it's probably clearer for people adding to `yaml` files to the repo. On the other hand, I prefer more concise naming (e.g., `description`, `contact_name`, `contact_email`, etc.)

File: data/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# List of existing software
+
+This folder holds information about existing software developed by the ForeSITE group. To add a new software, please create a new yaml file (either copying one of the existing ones or using the template below) and add it to the list below. Executing the [../README.qmd](../README.qmd) quarto file will automatically update the list of software included in the README.md at the root level of the repository and update the `software.xlsx` file included in this folder.
```
Comment (apulsipher) [inline data/README.md:3]: I don't see a GHA for building the README automatically. Is that something we want? If so, either add that to this PR or create an issue for it. 

In either caste, I think it would be helpful to include the exact command a person needs to run to execute the README.qmd, in case someone wants to update this but doesn't know how.

**Review by apulsipher — COMMENTED** [review]
I've got a few questions, but otherwise this looks great!

File: data/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+```yaml
```
[the remark this answers is not in the document]
Comment (apulsipher) [reply data/README.md:5]: Also, I think we should have an empty yaml file either here or as `example.yml` that users can simply copy and paste to fill out. Otherwise they have to copy an existing file (such as your example here) and then delete each field one by one before filling it out.

File: README.qmd (hunk 1 above)
Comment (gvegayon) [reply README.qmd:33]: Those are direct translations from the original excel file. Just using some regex magic to make it machine readable. We could address this in a separate issue, but I agree; we need better names! Could you create the issue?

File: data/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (gvegayon) [reply data/README.md:3]: I think you are wrong :). I can't find the discussion, but my colleagues and I at CDC work with "single line paragraphs" when it comes to markdown files. GitHub is better at highlighting differences on single lines instead, and it is easier to maintain text when you have a single line, otherwise you will always find yourself editing where the breaks are when you update the text.

File: data/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (gvegayon) [reply data/README.md:5]: Could you do both?

File: data/arima_generalize_arima_vector_autoregression.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+license: TBD
+languages: Python
+audience_type: TBD
+required_expertise_to_use_tool: TBD
+type_of_tool: Epidemic Model - Forecasting
+type_of_data_input_needed: 'The objective of this study is to use multiple time series
+  data to predict weekly infection counts for mutiple virus. Timeframe: 12/2002-01/2024'
+link_to_web_page_documentation_optional: no pubication plan
+link_to_source_code_optional: .na.character
+reviewer: Andrew
```
[the remark this answers is not in the document]
Comment (gvegayon) [reply data/arima_generalize_arima_vector_autoregression.yml:17]: That's legacy. Could you add an issue indicating we need to clean the ymls after this PR is merged?

File: data/software.xlsx
[the remark this answers is not in the document]
Comment (gvegayon) [reply data/software.xlsx:1]: I've thought about it. Let's add an issue for it!

File: data/arima_generalize_arima_vector_autoregression.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (apulsipher) [reply data/arima_generalize_arima_vector_autoregression.yml:17]: Created #20 to address this

File: data/airborne_release_of_infectious_pathogens_simulator.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+type_of_data_input_needed: Exposure localized to release area.
+link_to_web_page_documentation_optional: 1-Toth D, Gundlapalli A, Schell W, Bulmahn
+  K, Walton T, Woods C, Coghill C, Gallegos F, Samore M, Adler F (2013). Quantitative
+  models of the dose-response and time course of inhalational anthrax in humans. PLoS
+  Pathog, 9(8), e1003555. https://doi.org/10.1371/journal.ppat.1003555.                          2-Bulmahn
+  K, Canella M, Coghill C, Gallegos F, Gundlapalli A, Schell W, Toth D, Walton T,
+  Woods C (2012). Final Supplementary Risk Assessment for the Boston University National
+  Emerging Infectious Diseases Laboratories, National Institutes of Health. https://www.bu.edu/neidl/files/2013/01/SFEIR-Volume-III.pdf.
+link_to_source_code_optional: .na.character
+reviewer: George
```
[the remark this answers is not in the document]
Comment (apulsipher) [reply data/airborne_release_of_infectious_pathogens_simulator.yml:27]: Created #20 to address this

File: data/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (apulsipher) [reply data/README.md]: We might have to agree to disagree on this one :) 
I've never had a problem editing line breaks, except where the breaks are in the middle of a sentence (forcing word wrapping) rather than at the end of the sentence (after the period/full-stop). Hopefully we can both agree that introducing line breaks within a sentence to enforce word wrapping is no good :)

That said, I can live with single-line paragraphs as long as we're consistent.

File: README_files/figure-commonmark/wordcloud-1.png
Comment (apulsipher) [reply README_files/figure-commonmark/wordcloud-1.png:1]: Created #21 to address this
